### PR TITLE
4 fix/cleanups

### PR DESCRIPTION
**Pull request by gandrewstone**
expose useful InterpretBool function, remove unnecessary cs_main lock on getinfo rpc, cleanup getblockstats, and add needed Android implementation of GetStrongRandBytes

### CODE REVIEW

File: src/blockstorage/blockstorage.cpp [hunk 1]
```diff
@@ -517,6 +517,8 @@ bool WriteUndoToDisk(const CBlockUndo &blockundo,
  */
 bool ReadUndoFromDisk(CBlockUndo &blockundo, const CDiskBlockPos &pos, const CBlockIndex *pindex)
 {
+    if (pindex == nullptr)
+        return error("Genesis block has no undo information");
```
Comment (Greg-Griffith) [inline src/blockstorage/blockstorage.cpp]: Even though we always check pindex before passing it into this function, this is a good check. However, the genesis block does have undo data. when pindex is nullptr it is the block before genesis block so this error message should be changed.

Comment (Greg-Griffith) [reply src/blockstorage/blockstorage.cpp]: for clarity, if you were to consider the undo data its own chain an, its "tip" would always be the blockchain tip - 1

File: src/util.h [hunk 2]
```diff
@@ -497,6 +497,14 @@ void UnsetArg(const std::string &strArg);
  */
 void SetBoolArg(const std::string &strArg, bool fValue);
 
+/*
+ * Convert string into true/false
+ *
+ * @param strValue String to parse as a boolean
+ * @return true or false
+ */
+bool InterpretBool(const std::string &strValue);
```
Comment (sickpig) [inline src/util.h:506]: I'm cool to expose this to the rest of the code base.

Do you have some use case in mind already?

File: qa/rpc-tests/rpc_getblockstats.py [hunk 3]
```diff
@@ -177,5 +177,14 @@ def run_test(self):
         assert_raises_rpc_error(-5, 'Block not found', self.nodes[0].getblockstats,
                                 '000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f')
 
+        # check genesis block stats
+        gb = self.nodes[0].getblock("0")
+        gbstats = self.nodes[0].getblockstats(gb["hash"])
```
Comment (sickpig) [inline qa/rpc-tests/rpc_getblockstats.py:182]: nit: you could directly pass block height to getblockstats

**Review by sickpig — APPROVED** [review]

**Pull request merged**